server/src/test/java/org/eclipse/lsp/cobol/usecases/TestGlobalVariables.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.usecases;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.eclipse.lsp.cobol.usecases.engine.UseCaseEngine;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.junit.jupiter.api.Test;
+
+import static org.eclipse.lsp.cobol.service.delegates.validations.SourceInfoLevels.ERROR;
+
+/**
+ * Test Global variables errors.
+ */
+class TestGlobalVariables {
+  private static final String GLOBAL_USAGE =
+      "       Identification Division.\n"
+          + "       Program-Id. 'P1'.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01 {$*FOO} pointer GLOBAL.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO}.\n"
+          + "       Identification Division.\n"
+          + "       Program-Id. 'P11'.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO}.\n"
+          + "       End Program 'P11'.\n"
+          + "       End Program 'P1'.\n"
+          + "       Identification Division.\n"
+          + "       Program-Id. 'P2'.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$FOO|1}.\n"
+          + "       End Program 'P2'.";
+
+  private static final String VARIABLE_SHADOWING =
+      "       Identification Division.\n"
+          + "       Program-Id. 'P1'.\n"
+          + "       Data Division.\n"
+          + "       Working-Storage Section.\n"
+          + "       01 {$*FOO} GLOBAL.\n"
+          + "        05 {$*BAR} POINTER.\n"
+          + "       Procedure Division.\n"
+          + "           DISPLAY {$BAR}.\n"
```

Comment:
Please, add a case to verify the behavior of shadowed variable

Reply:
Agree, it will be a good case, but I have no idea how to implement it. I mean how to check it and what to check?

Reply:
If I understood the shadowing rules correctly, referencing `FOO` in the nested program should cause the "Invalid definition" error

Reply:
Maybe we are thinking about different things:
```
PROGRAM-ID. MAIN.
01 FOO GLOBAL.
PROGRAM-ID. NESTED.
01 FOO. <local overwrite global>
DISPLAY FOO. <It's ok, we use local var>
```
```
PROGRAM-ID. MAIN.
01 FOO GLOBAL.
PROGRAM-ID. NESTED.
DISPLAY FOO. <It's ok, we use global var>
```
Both cases are the same.

Reply:
Ah, ok then. Just add `DISPLAY FOO.` in the nested program to check that it does not produce any errors.